iati/tests/test_validator.py

```diff
@@ -846,16 +833,67 @@ def test_validation_codelist_vocab_user_defined_with_uri_unreadable(self, schema
         assert iati.validator.is_valid(data, schema_sectors)
 
 
+class TestValidateRulesets(object):
```

Comment:
Need to ensure that attributes set during the creation of a `ValidationError` are sensible - in particular, `line_number` and `context` are not available at this point, so it should be tested that there is no placeholder value (an `AttributeError` should likely be raised).